### PR TITLE
Graphql/workflow execution query

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Added WorkflowExecutionType to graphql (as well as resolver)
Added SamplesWorkflowExecutionsResolver to allow fetching samples through SamplesWorkflowExecution objects
Added WorkflowExecutionMetadataResolver to allow fetching metadata from a WorkflowExecution

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Via the gui, submit a couple of pipelines so you have data to fetch.
2. Query for workflow executions.

```graphql
query getWEs {
  workflowExecutions {
    nodes {
      id
    }
  }
}
```

3. Query for a specific workflow execution.

```graphql
query getOneWE {
  node(
    id: "gid://irida/WorkflowExecution/2eba5816-f486-4cec-b49a-aa1e933cdc91"
  ) {
  	... on WorkflowExecution {
      id
      cleaned
      httpErrorCode
      metadata
      runId
      state
      project{
        id
        name
      }
      group{
        id
        name
      }
      submitter {
        id
        email
      }
      submitterId
      tags
      workflowEngine
      workflowEngineParameters
      workflowEngineVersion
      workflowParams
      workflowType
      workflowTypeVersion
      workflowUrl
      samplesWorkflowExecutions {
        edges {
          node {
            id
          }
        }
      }
    }
  }
}
```

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by ChrisHuynh333 — APPROVED** [review]
Looks good!

File: app/graphql/types/workflow_execution_type.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Types
+  # WorkflowExecutions Type
+  class WorkflowExecutionType < Types::BaseObject
+    implements GraphQL::Types::Relay::Node
+    description 'A workflow execution'
+
+    field :blob_run_directory, String, null: true, description: 'WorkflowExecution blob run directory'
```
Comment (ericenns) [inline app/graphql/types/workflow_execution_type.rb:9]: **nitpick (blocking):** Can you remove this field, we don't want to expose this over the API.

**Review by ericenns — CHANGES_REQUESTED** [review]
This is looking pretty good, would just like one field removed.

File: app/graphql/types/workflow_execution_type.rb (hunk 1 above)
Comment (JeffreyThiessen) [reply app/graphql/types/workflow_execution_type.rb]: Fixed in https://github.com/phac-nml/irida-next/pull/832/commits/6f79351d179108caa81252dc14a4e0032d61cd83

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**